vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -330,11 +331,44 @@ func (feeder *clusterStateFeeder) GarbageCollectCheckpoints() {
 // Fetch VPA objects and load them into the cluster state.
 func (feeder *clusterStateFeeder) LoadVPAs() {
 	// List VPA API objects.
-	vpaCRDs, err := feeder.vpaLister.List(labels.Everything())
+	allVpaCRDs, err := feeder.vpaLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("Cannot list VPAs. Reason: %+v", err)
 		return
 	}
+
+	// Define a findElement function to check if clusterState.RecommenderName is in the vpaCRD.Spec.RecommenderName
+	findElement := func(slice []string, val string) bool {
```

Comment:
Do you want to limit recommender to only one at this moment?

Reply:
Yes, the current solution can only support one recommender. Otherwise, two recommenders will overwrite the RecommendedPodResources.

Reply:
The logic to check if only one recommender is defined in VPA is added in the handler of the admission controller.